internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -2009,6 +2010,27 @@ func clusterCompleted(clusterInstall *hiveext.AgentClusterInstall, clusterDeploy
 	})
 }
 
+func lastInstallationPreparationFailed(clusterInstall *hiveext.AgentClusterInstall, installation_preparation_completion_status string, installation_preparation_completion_status_reason string) {
+	var condStatus corev1.ConditionStatus = corev1.ConditionFalse
```

Comment:
Is this function called in all cases? Including when installation is still in progress, and if it succeeds?
If yes, please rename to indicate what it is actually doing